agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/wasbootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -233,6 +235,15 @@ public static void overlayEnvVars(Configuration config) throws IOException {
         loadInstrumentationEnabledEnvVars(config);
     }
 
+    private static void overlayRpConfiguration(Configuration config, RpConfiguration rpConfiguration)  {
+        if (rpConfiguration.connectionString != null) {
```

Comment:
empty check? unless you want to let them disable java agent by setting an empty connection string?